library/pathname/glob_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_relative '../../spec_helper'
+require 'pathname'
+
+describe 'Pathname#glob' do
+  before :all  do
+    @dir = tmp ''
+    @file_1 = @dir + 'lib/ipaddr.rb'
+    @file_2 = @dir + 'lib/irb.rb'
+    @file_3 = @dir + 'lib/.hidden.rb'
+
+    touch @file_1
+    touch @file_2
+    touch @file_3
+  end
+
+  after :all do
+    rm_r @dir
+  end
+
+  it 'returns [] for no match' do
+    Pathname.glob(@dir + 'lib/*.js').should == []
+  end
+
+  it 'returns matching file paths' do
+    Pathname.glob(@dir + 'lib/*i*.rb').sort.should == [Pathname.new(@file_1), Pathname.new(@file_2)].sort
+  end
+
+  it 'returns matching file paths when a flag is provided' do
+    expected = [Pathname.new(@file_1), Pathname.new(@file_2), Pathname.new(@file_3)].sort
+    Pathname.glob(@dir + 'lib/*i*.rb', File::FNM_DOTMATCH).sort.should == expected
+  end
+
+  it 'returns matching file paths when supplied :base keyword argument' do
+    Pathname.glob('*i*.rb', base: @dir + 'lib').sort.should == [Pathname.new('ipaddr.rb'), Pathname.new('irb.rb')].sort
+  end
+
+  ruby_version_is ''...'2.7' do
+    it 'raises an ArgumentError when supplied a flag and :base keyword argument' do
+      -> {
+        Pathname.glob(@dir + 'lib/*i*.rb', File::FNM_DOTMATCH, base: 'lib')
+      }.should raise_error(ArgumentError, 'wrong number of arguments (given 3, expected 1..2)')
+    end
+
+    it "raises an ArgumentError when supplied a keyword argument other than :base" do
+      -> {
+        Pathname.glob('*i*.rb', foo: @dir + 'lib')
+      }.should raise_error(ArgumentError, 'unknown keyword: foo')
+    end
```

Comment:
This spec seems to work on all versions, so I think it can moved out of the `ruby_version_is` guard.

Reply:
So the message was different in the two versions I used 
`ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]` said `foo`
while
`ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]` said `:foo`.

I could leave it as two specs or remove the string matching. What do you think?

Reply:
I see, then you can use a Regexp for the second argument of raise_error:
```ruby
/unknown keyword: :?foo/
```

Reply:
that makes sense. done!